presto-common/src/main/java/com/facebook/presto/common/block/DictionaryBlock.java

```diff
@@ -248,7 +248,7 @@ public long getLogicalSizeInBytes()
         for (int i = 0; i < getPositionCount(); i++) {
             int position = getId(i);
             if (seenSizes[position] < 0) {
-                seenSizes[position] = dictionary.getRegionSizeInBytes(position, 1);
+                seenSizes[position] = dictionary.getRegionLogicalSizeInBytes(position, 1);
```

Comment:
This loop is duplicated between here and getRegionLogicalSizeInBytes. Consider refactoring to avoid copy-paste.

```
public long getLogicalSizeInBytes() {
   return getRegionLogicalSizeInBytes(0, getPositionCount());
}

public long getRegionLogicalSizeInBytes(int positionOffset, int length) {
   if (positionOffset == 0 && length == getPositionCount() && logicalSizeInBytes >= 0) {
      return logicalSizeInBytes;
   }

   ...loop


   if (positionOffset == 0 && length == getPositionCount()) {
      logicalSizeInBytes = sizeInBytes;
   }
   return sizeInBytes;
}
}
```